src/travel-details-screens/utils.ts

```diff
@@ -321,138 +322,93 @@ export function getSecondsRemainingToBookingDeadline(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-  );
-  return (
-    secondsRemainingToDeadline >= secondsInOneHour &&
-    secondsRemainingToDeadline > secondsInNumberOfDaysAvailable
-  );
-}
-
-export function getIsTooLateToBook(
-  bookingArrangements: BookingArrangementFragment,
-  aimedStartTime: string,
-  now: number,
-): boolean {
-  const secondsRemainingToDeadline = getSecondsRemainingToBookingDeadline(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-  );
-  return secondsRemainingToDeadline < 0;
-}
-
-export function getBookingIsAvailableImminently(
-  bookingArrangements: BookingArrangementFragment,
-  aimedStartTime: string,
-  now: number,
-  flex_booking_number_of_days_available: number,
-): boolean {
-  const secondsRemainingToAvailable = getSecondsRemainingUntilBookingAvailable(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-    flex_booking_number_of_days_available,
-  );
-  return (
-    secondsRemainingToAvailable > 0 &&
-    secondsRemainingToAvailable < secondsInOneHour
-  );
-}
-
-export function getIsBookingAvailable(
+/**
+ * Get bookings status based on the given booking arrangements and aimed start
+ * time. Will return one of these values:
+ * - 'none': When there is no booking arrangements
+ * - 'early': It is too early to book the trip
+ * - 'bookable': The trip may be booked
+ * - 'late': The booking deadline has passed
+ * '´
+ * @param bookingArrangements The booking arrangements for the leg or estimated
+ * call
+ * @param aimedStartTime the aimed start time of the leg or estimated call
+ * @param now The now value to use, if not given the Date.now() is used
+ * @param flex_booking_number_of_days_available Number of days before latest
+ * booking time the user may book, if not given then the returned status can
+ * never be 'early'.
+ */
+export const getBookingStatus = (
   bookingArrangements: BookingArrangementFragment | undefined,
   aimedStartTime: string,
-  now: number,
-  flex_booking_number_of_days_available: number,
-): boolean {
-  if (!bookingArrangements) return false;
+  now: number = Date.now(),
+  flex_booking_number_of_days_available?: number,
+): BookingStatus => {
+  if (!bookingArrangements) return 'none';
 
-  const isTooEarly = getIsTooEarlyToBook(
-    bookingArrangements,
-    aimedStartTime,
-    now,
-    flex_booking_number_of_days_available,
-  );
-  const isTooLate = getIsTooLateToBook(
+  const secondsToDeadline = getSecondsRemainingToBookingDeadline(
     bookingArrangements,
     aimedStartTime,
     now,
   );
 
-  return !isTooEarly && !isTooLate;
-}
+  if (secondsToDeadline < 0) {
+    return 'late';
+  }
 
-export const getTripPatternBookingsRequiredCount = (tp: TripPattern): number =>
-  tp?.legs?.filter((leg) => !!leg.bookingArrangements).length;
+  if (flex_booking_number_of_days_available) {
+    const secondsBeforehandItCanBeBooked =
+      flex_booking_number_of_days_available * 24 * secondsInOneHour;
```

Comment:
Not sure how critical it is, but will this be off one hour twice a year for summer time / winter time?

Reply:
Yeah, but probably not critical. And there is the same issue with tickets, where a 24hour ticket is actually 24 hours, and not from 3pm to 3pm if there is a summer/winter time switch.